pkg/addonfactory/helm_agentaddon.go

```diff
@@ -171,8 +175,14 @@ func (a *HelmAgentAddon) getValues(
 
 	overrideValues = MergeValues(overrideValues, builtinValues)
 
-	values, err := chartutil.ToRenderValues(a.chart, overrideValues,
-		a.releaseOptions(cluster, addon), a.capabilities(cluster, addon))
+	var capabilities *chartutil.Capabilities
+	if hostingCluster == nil {
```

Comment:
I think we should have another variable to define version rather than the kubeVersion, you may need to check both kube version on mc and hosted cluster. Since manifests could be deployed on both clusters.